src/mlpack/core/data/imputation_methods/median_imputation.hpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * @file median_imputation.hpp
+ * @author Keon Kim
+ *
+ * Definition and Implementation of the MedianImputation class.
+ */
+#ifndef MLPACK_CORE_DATA_IMPUTE_STRATEGIES_MEDIAN_IMPUTATION_HPP
+#define MLPACK_CORE_DATA_IMPUTE_STRATEGIES_MEDIAN_IMPUTATION_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace data {
+/**
+ * This is a class implementation of simple median imputation.
+ * replace missing value with middle or average of middle values
+ * @tparam T Type of armadillo matrix
+ */
+template <typename T>
+class MedianImputation
+{
+ public:
+  /**
+   * Impute function searches through the input looking for mappedValue and
+   * replaces it with the median of the given dimension. The result is saved
+   * to the output.
+   *
+   * @param input Matrix that contains mappedValue.
+   * @param output Matrix that the result will be saved into.
+   * @param mappedValue Value that the user wants to get rid of.
+   * @param dimension Index of the dimension of the mappedValue.
+   * @param columnMajor State of whether the input matrix is columnMajor or not.
+   */
+  void Impute(const arma::Mat<T>& input,
+              arma::Mat<T>& output,
+              const T& mappedValue,
+              const size_t dimension,
+              const bool columnMajor = true)
+  {
+    //initiate output
+    output = input;
+    Impute(output, mappedValue, dimension, columnMajor);
+  }
+
+  /**
+   * Impute function searches through the input looking for mappedValue and
+   * replaces it with the median of the given dimension. The result is
+   * overwritten to the input matrix.
+   *
+   * @param input Matrix that contains mappedValue.
+   * @param mappedValue Value that the user wants to get rid of.
+   * @param dimension Index of the dimension of the mappedValue.
+   * @param columnMajor State of whether the input matrix is columnMajor or not.
+   */
+  void Impute(arma::Mat<T>& input,
+              const T& mappedValue,
+              const size_t dimension,
+              const bool columnMajor = true)
```

Comment:
> I think non static function is better for this case, because we can store the state of imputation in the class if we need to.

I agree in general, I only meant that we could mark this particular function static when it is implemented in classes that do not have any state.  I think `CustomImputation` is the only class where these methods could not be marked static.